compilesketches/compilesketches.py

```diff
@@ -893,13 +894,22 @@ def compile_sketch(self, sketch_path, clean_build_cache):
         if clean_build_cache:
             for cache_path in pathlib.Path("/tmp").glob(pattern="arduino*"):
                 shutil.rmtree(path=cache_path)
-
+        start_time = time.monotonic()
         compilation_data = self.run_arduino_cli_command(
             command=compilation_command, enable_output=self.RunCommandOutput.NONE, exit_on_failure=False)
+        diff_time = time.monotonic() - start_time
+        time_summary = ""
+        if diff_time > 60:
+            if diff_time > 360:
+                time_summary += f"{int(diff_time / 360)} hours "
+            time_summary += f"{int(diff_time / 60) % 60} minutes "
+        time_summary += f"{int(diff_time) % 60} seconds."
+
         # Group compilation output to make the log easy to read
         # https://github.com/actions/toolkit/blob/master/docs/commands.md#group-and-ungroup-log-lines
         print("::group::Compiling sketch:", path_relative_to_workspace(path=sketch_path))
         print(compilation_data.stdout)
+        print("Compilation time elapsed", time_summary)
```

Comment:
Make this consistent with the other compilation data.

For example:

https://github.com/2bndy5/compile-sketches/runs/6062736376?check_suite_focus=true#step:3:283

```text
  Compilation time elapsed 1 seconds.
Change in flash: 0 (0.0%)
Change in RAM for global variables: 0 (0.0%)
Change in compiler warning count: 0
```

- Move the print outside of the log group (the log group is intended to contain the Arduino CLI output alone)
- Use the same format as the other compilation data (e.g., `Compilation time elapsed: 1 s`)

Unlike the deltas data, I think it should be printed after each compilation, rather than only after the completion of the base compilation. The reason is because if it was only printed after the base compilation I think it would not be clear whether it was referring to the duration of the head compilation, the base compilation, or the sum of the two compilations.

Reply:
output is now [like so](https://github.com/2bndy5/compile-sketches/runs/6063691760?check_suite_focus=true#step:3:137)
```
> Compiling sketch: .github/workflows/testdata/sketches/BareMinimum
Compilation time elapsed 0s
Compiling previous version of sketch to determine memory usage change
> Compiling sketch: .github/workflows/testdata/sketches/BareMinimum
Compilation time elapsed 0s
Change in flash: 0 (0.0%)
Change in RAM for global variables: 0 (0.0%)
```
not sure if that's what you want though.

Reply:
That is it!